src/comm/TcpAcceptor.cc

```diff
@@ -369,7 +368,10 @@ Comm::TcpAcceptor::oldAccept(Comm::ConnectionPointer &details)
     // Sync with Comm ASAP so that abandoned details can properly close().
     // XXX : these are not all HTTP requests. use a note about type and ip:port details->
     // so we end up with a uniform "(HTTP|FTP-data|HTTPS|...) remote-ip:remote-port"
-    Descriptor sock(rawSock, FD_SOCKET, "HTTP Request");
+    const auto sock = rawSock;
+    fd_open(sock, FD_SOCKET, "HTTP Request");
+    details->fd = sock;
+    details->enterOrphanage();
```

Comment:
The above change brings us back to where this code was before my buggy commit https://github.com/squid-cache/squid/commit/7185c9e7dabe76fce29ea527c52b6053e53493a5. Instead of using Comm::Descriptor to delay "opening" of the half-baked `details` Connection (to avoid "leaking" an open but unusable Connection object to subscribers), we now close such connections in acceptOne() while correctly handling both explicit exceptions (i.e. former COMM_ERROR cases) and implicit ones (broken before that buggy commit).